aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -210,23 +209,24 @@ module aptos_framework::object {
         create_object_internal(creator_address, obj_addr, false)
     }
 
-    /// Create a new object from a GUID generated by an account.
-    public fun create_object_from_account(creator: &signer): ConstructorRef {
-        let guid = account::create_guid(creator);
-        create_object_from_guid(signer::address_of(creator), guid)
+    /// Create a new object by generating a random UUID based on transaction hash.
+    /// The created object is deletable as we can guarantee the same UUID can 
+    /// never be regenerated with future txs.
+    public fun create_object(creator: &signer): ConstructorRef {
+        let uuid = aptos_framework::transaction_context::create_uuid();
+        create_object_internal(signer::address_of(creator), uuid, true)
```

Comment:
Please add a domain separation here (as a suffix to the generated uuid)

Reply:
UUID has an address type (32 bytes), should instead of appending to UUID, can I replace the last byte with `0xFA`?

Reply:
I don't think we need to do anything here, since we add Scheme::DeriveUuid already in native code.

Reply:
Yep. AFAICT, no need to.

Reply:
It makes more sense IMO to have 0xFA here in the object code since all these 0xF something domain separations are object domain separators. Another benefits of this is to simplify the prediction or off-chain determination of tx-hash based uuids as they'd always just simply hash(txhash + creation index in tx)

Reply:
@movekevin this function will be used in other places/new applications as well. Having it separated from other schemes as well is useful (i.e. DeriveResourceAccountAddress and Ed25519 are non-object schemes)

And Scheme (from types/src/transaction/authenticator.rs) seems like a right place to track all the different schemes.
It feels like object.move is lacking, i.e. it has listed only subset of schemes, and can create conflict with Scheme, and should have a back-reference to Scheme enum.

But that's just my understanding, @alinush knows better how these were designed

Reply:
Well, it's a bit of a hodge-podge unfortunately... Here's a short history lesson.

1. We had a domain separation bug a while ago between MultiEd25519 addresses and resource account addresses.
2. We as a result added a domain separator to resource account address derivation **in Move**.
3. But, in order to remember this domain separator exists, we duplicated it in in Rust in the `Scheme` struct in `types/src/transaction/authenticator.rs`
    a. i.e., we added `DeriveResourceAccountAddress` and you'll notice it's **never used in Rust**
4. Then, when Aptos Objects were proposed, some new address derivation code was introduced there too.
5. As a result, a new `DeriveObjectId` separator was added to `Scheme` too. Once again, this is **never used in Rust**; just kept track of.
6. Incidentally, this code had a small, [non-critical domain-separation bug](https://github.com/aptos-labs/aptos-core/pull/6481) too.
7. After fixing it, we replaced `DeriveObjectId` with two other domain separators in `Scheme`: `DeriveObjectIdFromGuid` and `DeriveObjectIdFromSeed`; also **never used in Rust**

Now, we are adding another domain separator there called `DeriveUuid`, which is actually used **in Rust** to derive a (domain-separated) address.